app/Models/Foundation/Summit/Speakers/PresentationSpeakerAssignment.php

```diff
@@ -0,0 +1,121 @@
+<?php namespace App\Models\Foundation\Summit\Speakers;
+/*
+ * Copyright 2022 OpenStack Foundation
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+use App\Models\Foundation\Main\IOrderable;
+use Doctrine\ORM\Mapping AS ORM;
+use App\Models\Utils\BaseEntity;
+use models\summit\Presentation;
+use models\summit\PresentationSpeaker;
+use models\utils\One2ManyPropertyTrait;
+
+/**
+ * @ORM\Entity
+ * @ORM\Table(name="Presentation_Speakers")
+ * Class PresentationSpeakerAssignment
+ * @package models\summit
+ */
+class PresentationSpeakerAssignment extends BaseEntity implements IOrderable
+{
+    use One2ManyPropertyTrait;
+
+    protected $getIdMappings = [
+        'getSpeakerId' => 'speaker',
+        'getPresentationId' => 'presentation',
+    ];
+
+    protected $hasPropertyMappings = [
+        'hasSpeaker' => 'speaker',
+        'hasPresentation' => 'presentation',
+    ];
+
+    /**
+     * @ORM\ManyToOne(targetEntity="models\summit\PresentationSpeaker", inversedBy="presentations_assignment", fetch="EXTRA_LAZY")
+     * @ORM\JoinColumn(name="PresentationSpeakerID", referencedColumnName="ID", onDelete="SET NULL")
+     * @var PresentationSpeaker
+     */
+    private $speaker;
+
+    /**
+     * @ORM\ManyToOne(targetEntity="models\summit\Presentation", inversedBy="speakers_assignment", fetch="EXTRA_LAZY")
+     * @ORM\JoinColumn(name="PresentationID", referencedColumnName="ID", onDelete="SET NULL")
+     * @var Presentation
+     */
+    private $presentation;
+
+    /**
+     * @ORM\Column(name="`Order`", type="integer")
```

Comment:
please another column name ORM could potentially generate SQL raw queries that are illegal bc Order is a reserved word

Reply:
keep in mind that the name field is "backtick protected" so shouldn't be an issue. This nomenclature was already used in other parts of the code.